src/bin/cargo/commands/add.rs

```diff
@@ -188,12 +192,21 @@ pub fn exec(config: &mut Config, args: &ArgMatches) -> CliResult {
 
     let dependencies = parse_dependencies(config, args)?;
 
+    let ignore_rust_version = args.flag("ignore-rust-version");
+    if ignore_rust_version {
+        config
+            .cli_unstable()
+            .fail_if_stable_opt("ignore-rust-version", 0)?;
```

Comment:
> Note: fail_if_stable_opt currently has its tracking issue set to 0. We can either pick an issue for that, or just write our own error message that does not include a link to an issue. Of course, don't merge until this has been resolved.

One option is that this PR doesn't close out #10653 but we keep it open to track to stabilization, so it acts as its own tracking issue.